tests/test_counting_hash.py

```diff
@@ -212,6 +212,135 @@ def test_simple_median():
     assert int(stddev * 100) == 50        # .5
 
 
+def test_filter_on_median():
+    hi = khmer.new_counting_hash(6, 1e6, 2)
+
+    hi.consume("AAAAAA")
+    assert hi.filter_on_median("AAAAAA", 1)
+    assert hi.filter_on_median("AAAAAA", 2) is False
+
+    hi.consume("AAAAAA")
+    assert hi.filter_on_median("AAAAAA", 2)
+    assert hi.filter_on_median("AAAAAA", 3) is False
+
+    hi.consume("AAAAAA")
+    assert hi.filter_on_median("AAAAAA", 3)
+    assert hi.filter_on_median("AAAAAA", 4) is False
+
+    hi.consume("AAAAAA")
+    assert hi.filter_on_median("AAAAAA", 4)
+    assert hi.filter_on_median("AAAAAA", 5) is False
+
+    hi.consume("AAAAAA")
+    assert hi.filter_on_median("AAAAAA", 5)
+    assert hi.filter_on_median("AAAAAA", 6) is False
+
+
+def test_filter_on_median_single_gt():
+    K = 20
+    hi = khmer.new_counting_hash(K, 1e6, 2)
+
+    kmers = ['ATCGATCGATCGATCGATCG',
+             'GTACGTACGTACGTACGTAC',
+             'TTAGTTAGTTAGTTAGTTAG']
+
+    for kmer in kmers:
+        hi.consume(kmer)
+        assert hi.filter_on_median(kmer, 1) is True
+
+
+def test_filter_on_median_single_lt():
+    K = 20
+    hi = khmer.new_counting_hash(K, 1e6, 2)
+
+    kmers = ['ATCGATCGATCGATCGATCG',
+             'GTACGTACGTACGTACGTAC',
+             'TTAGTTAGTTAGTTAGTTAG']
+
+    for kmer in kmers:
+        hi.consume(kmer)
+        assert hi.filter_on_median(kmer, 2) is False
+
+
+def test_filter_on_median_odd_gt():
+    K = 20
+    hi = khmer.new_counting_hash(K, 1e6, 2)
+
+    seqs = ['ATCGATCGATCGATCGATCGCC',
+            'GTACGTACGTACGTACGTACCC',
+            'TTAGTTAGTTAGTTAGTTAGCC']
+
+    for seq in seqs:
+        hi.consume(seq)
+        assert hi.filter_on_median(seq, 1) is True
+
+
+def test_filter_on_median_odd_lt():
+    K = 20
+    hi = khmer.new_counting_hash(K, 1e6, 2)
+
+    seqs = ['ATCGATCGATCGATCGATCGCC',
+            'GTACGTACGTACGTACGTACCC',
+            'TTAGTTAGTTAGTTAGTTAGCC']
+
+    for seq in seqs:
+        hi.consume(seq)
+        assert hi.filter_on_median(seq, 2) is False
+
+
+def test_filter_on_median_even_gt():
+    K = 20
+    hi = khmer.new_counting_hash(K, 1e6, 2)
+
+    seqs = ['ATCGATCGATCGATCGATCGCCC',
+            'GTACGTACGTACGTACGTACCCC',
+            'TTAGTTAGTTAGTTAGTTAGCCC']
+
+    for seq in seqs:
+        hi.consume(seq)
+        assert hi.filter_on_median(seq, 1) is True
+
+
+def test_filter_on_median_even_lt():
```

Comment:
please annotate the difference from 'test_filter_on_median_odd_lt' (seqs are different lengths)